### PR TITLE
Rov 76 validate the fields using typescript

### PR DESCRIPTION
**Pull request by Varuno8**

### CODE REVIEW

File: src/app/application/registration-1/registration-1.component.html [hunk 1]
```diff
@@ -19,7 +25,64 @@ <h1>{{ title }}</h1>
...
+        <label for="birthDate">Birth Date:</label>
+        <div class="input-group">
+          <input
+            type="date"
+            class="form-control"
+            ngbDatepicker
+            #d="ngbDatepicker"
+            formControlName="birthDate"
+          />
+          <div *ngIf="userForm.get('birthdate').hasError('lessthen87')">
```
Comment (rachittyagi) [inline src/app/application/registration-1/registration-1.component.html:65]: max date error message is missing

File: src/app/application/registration-1/registration-1.component.html [hunk 2]
```diff
@@ -19,7 +25,64 @@ <h1>{{ title }}</h1>
         <div *ngIf="userForm.get('name').hasError('pattern')">
           Name should only contain alphabets.
         </div>
+
+        <label for="email">Email</label>
+        <input
```
Comment (rachittyagi) [inline src/app/application/registration-1/registration-1.component.html:30]: error message is missing for email validation

File: src/app/application/registration-1/registration-1.component.html [hunk 3]
```diff
@@ -19,7 +25,64 @@ <h1>{{ title }}</h1>
...
+          class="form-control"
+          type="text"
+          placeholder="Enter email"
+          id="email"
+          formControlName="email"
+        />
+
+        <label for="phoneNumber">Phone Number</label>
+        <input
```
Comment (rachittyagi) [inline src/app/application/registration-1/registration-1.component.html:39]: validation error message for this field is also missing

File: src/app/application/registration-1/registration-1.component.html [hunk 4]
```diff
@@ -19,7 +25,64 @@ <h1>{{ title }}</h1>
...
+          class="form-control"
+          type="tel"
+          placeholder="Phone Number"
+          id="phoneNumber"
+          formControlName="phoneNumber"
+        />
+
+        <label for="password">Password</label>
+        <input
```
Comment (rachittyagi) [inline src/app/application/registration-1/registration-1.component.html:48]: validation error message for this field is also missing

File: src/app/application/registration-1/registration-1.component.html [hunk 5]
```diff
@@ -19,7 +25,64 @@ <h1>{{ title }}</h1>
...
+        >
+        </ng-select>
+        <div *ngIf="userForm.get('country').hasError('invalidCountryId')">
+          Invalid Country ID.
+        </div>
+
       </div>
     </div>
   </div>
+  <button type="button" class="btn btn-primary" (click)="submitForm()">
```
Comment (rachittyagi) [inline src/app/application/registration-1/registration-1.component.html:85]: disable the button until all fields are validated

**Review by rachittyagi — CHANGES_REQUESTED** [review]

File: src/app/application/registration-1/registration-1.component.html (hunk 2 above)
Comment (Varuno8) [reply src/app/application/registration-1/registration-1.component.html:30]: corrected the mistake

File: src/app/application/registration-1/registration-1.component.html (hunk 4 above)
Comment (Varuno8) [reply src/app/application/registration-1/registration-1.component.html:48]: corrected the mistake

File: src/app/application/registration-1/registration-1.component.html (hunk 1 above)
Comment (Varuno8) [reply src/app/application/registration-1/registration-1.component.html:65]: corrected the mistake

File: src/app/application/registration-1/registration-1.component.html (hunk 5 above)
Comment (Varuno8) [reply src/app/application/registration-1/registration-1.component.html:85]: corrected the mistake

File: src/app/application/registration-1/registration-1.component.html [hunk 6]
```diff
@@ -19,7 +25,82 @@ <h1>{{ title }}</h1>
...
+        <label for="birthDate">Birth Date:</label>
+        <div class="input-group">
+          <input
+            type="date"
+            class="form-control"
+            ngbDatepicker
+            #d="ngbDatepicker"
+            formControlName="birthDate"
+          />
+          <div *ngIf="userForm.get('birthDate').hasError('required')">
```
Comment (rachittyagi) [inline src/app/application/registration-1/registration-1.component.html:76]: formControlName should be exact same as in reactive form declaration.

File: src/app/application/registration-1/registration-1.component.html [hunk 7]
```diff
@@ -19,7 +25,82 @@ <h1>{{ title }}</h1>
...
+            Birth Date is required.
+          </div>
+          <div *ngIf="userForm.get('birthdate').hasError('lessthen87')">
```
Comment (rachittyagi) [inline src/app/application/registration-1/registration-1.component.html:79]: there should be check for max date as well, future date should not be selectable

**Review by rachittyagi — CHANGES_REQUESTED** [review]

**Review by rachittyagi — APPROVED** [review]

**Pull request merged**